src/js/components/utils.tsx

```diff
@@ -36,6 +41,105 @@ import {
 import routes from '@/utils/routes';
 import githubIcon from '#/github.svg';
 
+interface Crumb {
+  name: string;
+  url?: string;
+}
+
+export const DetailPageLayout = ({
+  title,
+  description,
+  repoUrl,
+  breadcrumb,
+  onRenderHeaderActions,
+  sidebar,
+  children,
+}: {
+  title: string;
+  description?: string;
+  repoUrl: string;
+  breadcrumb: Crumb[];
+  onRenderHeaderActions?: () => JSX.Element;
+  sidebar?: ReactNode;
+  children?: ReactNode;
+}) => {
+  const descriptionHasTitle =
+    description &&
+    (description.startsWith('<h1>') || description.startsWith('<h2>'));
+
+  return (
+    <>
+      <PageHeader
+        className="page-header slds-p-around_x-large"
+        title={title}
+        info={<RepoLink url={repoUrl} shortenGithub />}
+        onRenderControls={onRenderHeaderActions}
+      />
+      <div
+        className="slds-p-horizontal_x-large
+          slds-p-top_x-small
+          ms-breadcrumb"
+      >
+        <BreadCrumb
+          trail={[
```

Comment:
I don't like this interface, but I recognize that it's not under our control, so :(

Reply:
What part of it don't you like? We can control how the array is constructed, but we don't have control over the format of the eventual array.

This was the quick-and-dirty way to start with a common item, then add custom ones. I'm open to alternative suggestions.

Reply:
I would love a breadcrumb that just accepted an array and formatted it smartly, but I get that breadcrumb doesn't and this is _fine_ given those constraints.